chalk-integration/src/lowering.rs

```diff
@@ -1448,6 +1460,36 @@ impl LowerTy for Ty {
     }
 }
 
+trait LowerConst {
+    fn lower(&self, env: &Env) -> LowerResult<chalk_ir::Const<ChalkIr>>;
+}
+
+impl LowerConst for Const {
+    fn lower(&self, env: &Env) -> LowerResult<chalk_ir::Const<ChalkIr>> {
```

Comment:
I'm surprised we didn't already have something like this code?

Reply:
It wasn't needed before, because we only had consts in generic arg positions and parser didn't have a notion of `Const`; it was flattened into `GenericArg`.